src/scripts/gym/GymClearReward.ts

```diff
@@ -0,0 +1,29 @@
+class GymClearReward {
+    constructor(
+        public clearNumber: number,
+        public rewardFunction: (gym: Gym) => void
+    ) {}
+
+    public isRewardClear(clear: number): boolean {
+        return this.clearNumber === clear;
+    }
+}
+
+class RepeatingGymClearReward extends GymClearReward {
+    constructor(
+        startClearNumber: number,
+        private repeatPattern: number,
+        private repeatUntilClear: number,
```

Comment:
I would recommend default values :
```suggestion
        private repeatPattern: number,
        startClearNumber = 0,
        private repeatUntilClear = Infinity,
```
That way, default value for `startClearNumber` allows `repeatPattern` to be the first clear at which the pattern starts, and default value for `repeatUntilClear` makes the check `this.repeatUntilClear > 0` unnecessary in function `isRewardClear`.